content/lectures/lec05.md

```diff
@@ -3,8 +3,149 @@ title: "Practical Kubernetes"
+
+A PV is really a representation of the physical block where storage exists, and the PVC is a request to access that storage. This feels a little over-engineered, but it is actually really helpful when using cloud services like AWS where volumes are allocated independently of the machine running the containers.
+
+This is one of the key components of Kubernetes that lets us treat our serves as cattle, not pets. When volumes and storage are separated from our application code, we're able to separate our concerns around running our code from persisting our data. 
+
+## StatefulSet
+
+[StatefulSet](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/) is a deployment where each pod gets a persistent volume. This is used for applications that need to be resistant to reboots. Common applications include:
+- Distributed Databases: MongoDB, CockroachDB, Cassandra
+- Games: Minecraft
+- Search: AWS Elasticsearch
+
+![StatefulSet Diagram][/img/lec05/statefulset.png]
+
+Note that each StatefulSet has its own PersistentVolume, and so writes to PersistentVolume 0 from Pod 0 in the diagram will not be readable from either Pod 1 or Pod 2. The application itself will need to coordinate the state of the PersistentVolumes using some kind of replication strategy.
+
+## DaemonSet
+
+A [DaemonSet](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/) is a deployment which has one pod running on each node in the cluster. These are used to provide node-wide services like authentication gateways, ingress routers, and log collectors.
+
+![DaemonSet Diagram][/img/lec05/daemonset.png]
+
+## Annotations
+
+[Annotations](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/) aren't resources on their own, but a type of resource metadata used to store auxillary key/value information about a resource. Annotations aren't used by Kubernetes itself, but other software in and around your cluster might use the extra metadata to determine its own behavior. 
+
+## Kubernetes as an Abstraction
+
+(@peyton or @armaan can you rewrite this paragraph? not sure i'm gonna phrase it better than y'all)
+Sometimes instead of thinking about Kubernetes as a container orchestration tool, it can be better to think about it as an abstraction that allows us to reason about infrastructure. Kubernetes is the most popular container orchestration tool not because it is the best at orchestrating containers, but because the abstractions are simple and elegant.
```

Comment:
i'll rewrite along with the demo pass. just move the note into a markdown comment for now